lib/utilities/firebase_service.dart

```diff
@@ -45,17 +45,28 @@ class FirebaseService {
 
   // Register a user for an event
   Future<void> registerForEvent(String eventId, String userId) async {
-    try {
-      final eventDoc = _firestore.collection('events-collection').doc(eventId);
-      await eventDoc.update({
-        'registeredUsers': FieldValue.arrayUnion([userId]),
-        'slotsAvailable': FieldValue.increment(-1), // Decrement available slots
-      });
-      print('User $userId registered for event $eventId');
-    } catch (e) {
-      print('Error registering for event: $e');
-    }
+  try {
+    final eventDoc = _firestore.collection('events-collection').doc(eventId);
+
+    // Add user to the event's registered users list and decrement available slots
+    await eventDoc.update({
+      'registeredUsers': FieldValue.arrayUnion([userId]),
```

Comment:
If you have the registration collection, then this is unnecessary, you're saving the same data twice.